controllers/actual_lrp_lifecycle_controller.go

```diff
@@ -204,17 +204,17 @@ func (h *ActualLRPLifecycleController) CrashActualLRP(ctx context.Context, logge
 		return nil
 	}
 
-	desiredLRP, err := h.desiredLRPDB.DesiredLRPByProcessGuid(ctx, logger, actualLRPKey.ProcessGuid)
-	if err != nil {
-		logger.Error("failed-fetching-desired-lrp", err)
+	schedInfos, err := h.desiredLRPDB.DesiredLRPSchedulingInfos(ctx, logger, models.DesiredLRPFilter{ProcessGuids: []string{actualLRPKey.ProcessGuid}})
```

Comment:
What about exposing a method like DesiredLRPSchedulingInfoByProcessGuid in the [desired_lrp_db](https://github.com/cloudfoundry/bbs/blob/0007fcda733509f73717fa966d8dcd8ff92ce62c/db/desired_lrp_db.go) interface, so that we can get rid of the array length validation?